flax/linen/module.py

```diff
@@ -538,6 +538,80 @@ def nowrap(fun: _CallableT) -> _CallableT:
   return fun
 
 
+def nontransparent(fun: _CallableT) -> _CallableT:
```

Comment:
Thinking about this a bit more high level: What happens actually if you do `module.init(..., method=Foo.up)`? Like, is it an actual compact method? If not, then this should probably be explained clearly. If so, then this beg the question: why don't we just extend `nn.compact` to allow multiple methods? I think there were good reasons to keep one method and it probably isn't wise to change this behavior, but the difference between the proposed decorator and nn.compact isn't really clear to me.

Reply:
`module.init(..., method=Foo.up)` should behave like a compact method BUT the name `up` would be part of the param structure which is the point here.

Reply:
Alright, thanks for clarifying. Maybe the name could then be something like `@nn.compact_name_scope`?

Also, would it still work if someone would only create methods with this decorator, or do we always need one method annotated with `@nn.compact`?

Reply:
I like that name better!

> Also, would it still work if someone would only create methods with this decorator, or do we always need one method annotated with @nn.compact?

This implementation does NOT require you to have a main `compact` method 🙂

Reply:
I'll update the PR soon with the new name.